### DISCUSSION TITLE
All Glorys data available on uda up to October 2024, Glofas available by end of day, Era5 by next week

### DISCUSSION THREAD

**Discussion by uwagura** [NOAA-CEFI-Regional-Ocean-Modeling/seasonal-workflow#5] [General]
Hi all. Just a few updates on the UDA

1.  All Glorys reanalysis data form 01/1993 to 09/2024 is now available in `/uda/Global_Ocean_Physics_Reanalysis/global/daily/` for all available variables.
2. Glorys Analysis data from 09/25/2024 to 10/31/2024 is now available in `/uda/CEFI/GLOBAL_ANALYSISFORECAST_PHY_001_024/`
3. As I mentioned previously I currently have all GloFAS v3.1 from 1979-2024 data in my work directory (` /work/Utheri.Wagura/datasets/glofas/v3.1`), as well as the v4.0 data from 1995-2024 ( ` /work/Utheri.Wagura/datasets/glofas/v4.0`). I am meeting with Kris later today to discuss the best way to organize this data on the uda. I expect this will be relatively straightforward, so the Glofas v3.1 data should be on the uda by this afternoon as well
4. Kris is working on getting the ERA5 data requested by both CEFI and other groups onto the uda, and as of right now the plan is to have it there by next week. 

A few questions:

1. I initially planned on using copernicusmarine's default directory structure for the Glorys Analysis data to make syncs/file transfers easier, but then I realized that copernicusmarine groups `uo` and `vo` data together, so this would have made it harder to access these vars individually. I split these variables into their own directories following a similar structure to the Reanalysis data. Which directory structure do we want to use? This is our folder, so we can organize this data however we would like. 
2. I didn't post any Glorys Analysis data for November onto the UDA because when I initially did the downloads, this data was still forecast data, and not nowcast/hindcast data. Does anyone know if the analysis data changes signficanlty between these versions?

Let me know how the current data looks, and if there is anything else I should work on regarding this task!

**Comment by uwagura** [comment]
A quick update on this: the GloFAS v3.1 data can now be found in `/uda/GloFAS/LISFLOOD/consolidated/global/river_discharge/v3.1/`. This path may change based on discussions that Kris and I have, but the data should be somewhere in this dir no matter what. 

The only dataset left to push to the UDA is ERA5 ( and glofas v4.0, but I have most of this data in my work dir already). The one holdup with ERA5 is that we have been storing it as yearly files within the UDA ( ` /archive/uda/ERA5/Hourly_Data_On_Single_Levels/reanalysis/global/1hr-timestep/annual_file-range` ), but we will only need data from Jan of this year until the end of November for CEFI, so it probably doesn't make sense to push ERA5 to the path above. Should we put the ERA5 data in `/uda/CEFI` instead, and if so what format should we store it in (i.e daily files, monthly, etc.)? Let me know what you guys think.

**Reply by andrew-c-ross** [reply]
In the end we will probably `ncrcat` the ERA5 data together into one file per variable for the partial year, so it shouldn't matter too much what the raw files are saved as. Ideally, though, we eventually would automatically acquire the latest ERA5 data on a roughly daily basis, so keeping separate files for each day might make more sense.

**Comment by yichengt900** [comment]
> Hi all. Just a few updates on the UDA
> 
> 1. All Glorys reanalysis data form 01/1993 to 09/2024 is now available in `/uda/Global_Ocean_Physics_Reanalysis/global/daily/` for all available variables.
> 2. Glorys Analysis data from 09/25/2024 to 10/31/2024 is now available in `/uda/CEFI/GLOBAL_ANALYSISFORECAST_PHY_001_024/`
> 3. As I mentioned previously I currently have all GloFAS v3.1 from 1979-2024 data in my work directory (` /work/Utheri.Wagura/datasets/glofas/v3.1`), as well as the v4.0 data from 1995-2024 ( ` /work/Utheri.Wagura/datasets/glofas/v4.0`). I am meeting with Kris later today to discuss the best way to organize this data on the uda. I expect this will be relatively straightforward, so the Glofas v3.1 data should be on the uda by this afternoon as well
> 4. Kris is working on getting the ERA5 data requested by both CEFI and other groups onto the uda, and as of right now the plan is to have it there by next week.
> 
> A few questions:
> 
> 1. I initially planned on using copernicusmarine's default directory structure for the Glorys Analysis data to make syncs/file transfers easier, but then I realized that copernicusmarine groups `uo` and `vo` data together, so this would have made it harder to access these vars individually. I split these variables into their own directories following a similar structure to the Reanalysis data. Which directory structure do we want to use? This is our folder, so we can organize this data however we would like.
> 2. I didn't post any Glorys Analysis data for November onto the UDA because when I initially did the downloads, this data was still forecast data, and not nowcast/hindcast data. Does anyone know if the analysis data changes signficanlty between these versions?
> 
> Let me know how the current data looks, and if there is anything else I should work on regarding this task!

Hi @uwagura, thanks for the updates! I really appreciate all your effort. It looks like the Glorys Reanalysis has now been extended through 2024/10/01. I don’t have a clear answer to your first question, but from my experience with the Copernicus Marine CLI sync function,  it seems to work best if the local data structure mirrors the remote one, though I haven’t fully tested the sync capability myself.

I’d suggest we consider keeping the Glorys Analysis data structure aligned with the remote source and performing additional processing once it’s classified as `hcst`, making it easier for model inputs preparation, but of course it's up to you.

Although `hcst` data is preferred for analysis, there may be cases where we may need to use `nwct` or even `fcst` data for certain periods. For example, if now we’re aiming to start a seasonal prediction from 2024/11/01, only `nwct` data is available for 2024/10/30-31. I haven’t examined the differences between these datasets in detail. It might be worth looking into.
